Test addEventListener abort signal

benjamingr opened the pull request:
Tests for https://github.com/whatwg/dom/pull/919

benjamingr committed:
initial addEventListener signal tests

benjamingr commented:
I _think_ the WPTs are ready for review :]

annevk commented on dom/events/AddEventListenerOptions-signal.html:
```diff
@@ -0,0 +1,71 @@
+  const et = new EventTarget();
+  const controller = new AbortController();
+  et.addEventListener('test', handler, { signal: controller.signal, once: true });
+  controller.abort();
```
Why would this not remove it already? I'm not sure how this test works.

benjamingr replied:
This would remove it already, this test checks that even if I add an event listener with `once`, passing a signal and aborting the controller removes it :]

annevk replied:
I see, that makes sense. We should also test the inverse, that once still removes it if you pass a signal in.

benjamingr replied:
I'll add tests for that and cases you mentioned as well.

annevk replied:
It seems there's still no test that tests once itself. I.e., dispatch an event twice and only bring count up to 1?

annevk reviewed (commented):
Thanks for writing these, looks like a good start!

If you want to test interaction with once/capture it seems you should also check the event listener is still correctly invoked for them.

We should also test passing the signal to multiple listeners, both with different event names and the same event name. And there also check what happens if you abort from one of the listeners (which listeners will still get invoked and which will not).

You might also want to consider https://web-platform-tests.org/writing-tests/testharness.html#tests-for-other-or-multiple-globals-any-js to reduce the boilerplate and make these tests even more reusable across JavaScript hosts.

benjamingr commented:
Done, thanks for the review please take a look :]

annevk commented:
I think there should be more coverage for the multiple listener scenarios. E.g., multiple listeners for the same name and signal, but different handlers. And then one of the handlers aborts. The remaining handlers should not run (due to the removed field of their event listener being set).

benjamingr commented:
@annevk 

> , multiple listeners for the same name and signal, but different handlers. And then one of the handlers aborts.

Is that not [this test](https://github.com/web-platform-tests/wpt/pull/26472/files#diff-96216b6e1d920cc33c8560820458aeb2f57e707edcc1c290bf642163398aabd4R90-R103)?

smaug---- commented:
I'd like to see tests where one listener adds another listener with signal and then aborts it.
And similar but with nested event dispatch.

benjamingr commented:
@smaug---- thanks! I've added a test for your first suggestion (and rebased) . I'm not sure what you mean by "similar but with nested event dispatch".

benjamingr commented:
Just updating that I've added the "nested event dispatch" test after talking about it with @smaug----  and Anne on #whatwg

annevk commented:
Should we remove the test added in https://github.com/web-platform-tests/wpt/pull/26462 as part of this or incorporate it? If we keep it as-is we should remove `.tentative` from the filename.

benjamingr commented:
@annevk likely remove since it's tested in https://github.com/web-platform-tests/wpt/pull/26472/files#diff-96216b6e1d920cc33c8560820458aeb2f57e707edcc1c290bf642163398aabd4R15-R17 but there might be merit in a simpler tests that test a more basic use case?

annevk commented:
Not that I know of; let's remove it as part of this PR.

cc @josepharhar

benjamingr committed:
fixup! cr

benjamingr commented:
Done
